src/mumble/PipeWire.cpp

```diff
@@ -0,0 +1,437 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "PipeWire.h"
+
+#include "Global.h"
+
+#include <pipewire/core.h>
+#include <pipewire/keys.h>
+#include <pipewire/stream.h>
+
+#include <spa/param/audio/format-utils.h>
+#include <spa/pod/builder.h>
+
+#ifdef Q_CC_GNU
+#	define RESOLVE(var)                                                    \
+		{                                                                   \
+			var = reinterpret_cast< __typeof__(var) >(m_lib.resolve(#var)); \
+			if (!var)                                                       \
+				return;                                                     \
+		}
+#else
+#	define RESOLVE(var)                                                                     \
+		{                                                                                    \
+			*reinterpret_cast< void ** >(&var) = static_cast< void * >(m_lib.resolve(#var)); \
```

Comment:
Why do you assign via void pointers? That seems dangerous to me

If you use the `decltype` solution we should be able to use that for all compilers

Reply:
The returned data type is `QFunctionPointer`: https://doc.qt.io/qt-5/qlibrary.html#resolve